include/deal.II/lac/petsc_matrix_base.h

```diff
@@ -304,6 +304,12 @@ namespace PETScWrappers
      */
     MatrixBase();
 
+    /**
+     * Initialize a Matrix from a PETSc Mat object. Note that we do not copy
+     * the matrix.
```

Comment:
Could you add a short note about the reference counting here? Not everyone knows about that PETSc feature.

Reply:
All PETSc objects are plain C pointers behaving like `std::shared_ptr`